modules/blockbuilder/blockbuilder.go

```diff
@@ -206,7 +206,10 @@ func (b *BlockBuilder) starting(ctx context.Context) (err error) {
 		b.kadm,
 		b.logger,
 		b.cfg.IngestStorageConfig,
-		b.getAssignedPartitions)
+		b.getAssignedPartitions,
+		func() {
+			b.kafkaClient.ForceMetadataRefresh()
+		})
```

Comment:
You can pass the function directly without wrapping it.
```suggestion
		b.kafkaClient.ForceMetadataRefresh
```